### PR TITLE
Adding 'Generic Mod Config Menu' mod integration to resolve issue #42…

### PR DESCRIPTION
**Pull request by m4ny**
… in og repo.

## Summary

- The sound/visual effect and the notification can now be switched on or off by config file. 
- Integrated "[Generic Mod Config Menu](https://github.com/spacechase0/GenericModConfigMenu)" capabilities to your implementation. 

Therefore added 3 new boolean fields in the `ModsConfig.cs` class and altered the `OnGameLaunched` method in `ModEntry.cs`.
I used the [this example code](https://gist.github.com/spacechase0/2d8d4dbffe5f2ce9457d2c891a8b99e3/)  to implement this changes.

## Things to consider
1. I don't know if "old config files" without the new three boolean attributes will lead to problems. Maybee the fixup method must be expanded for this reason?
2. The screenshot rules information is not configurable with the menu. I simply wouldn't know how to implement this. A corresponding disclaimer has been included to the config menu as a label with a text hint.
3. You could probably add the constant default values as settable within range?
4. Is the `Thread.Sleep` operation at line 687 in `ModEntry.cs` necessary; or could it also be made configurable to set `MILLISECONDS_TIMEOUT` "dynamically" to potentially avoid lag everytime a screenshot is taken?
5. The smapi nugget package you're using could be updated.

## Testing
I compiled and ran the game without any problems. The newly added features seemed to be working. Due to the fact that this is my first mod edit and pull request, I just wanna' stress that you may take a closer look.

### CODE REVIEW

**Comment by CompSciLauren** [comment]
Hey, first off thank you so much for working on this and creating a PR! Fantastic stuff. I'm reviewing the changes and testing things out, I'll try to finish reviewing this soon.

Just wanted to respond to your "things to consider" section:
1. I tested this out and don't believe it will cause any problems. The config seems to update automatically without any issues.
2. The disclaimer is definitely good to include in this case, good idea. I think what we might do is just try to get these changes merged, and we can make a separate GitHub Issue to add the rest of the config options at a later time.
3. Sorry I don't think I understand what you're referring to here. Can you please explain a bit more? What constant default values are you referring to, and what do you mean by "settable within range"?
4. That sounds like a good idea, I'll have to look into that.
5. Good to know, thanks. I think I'll wait until we get this PR merged so that we can get these changes out to users. But after that I'll check that out. Logged #47 to track this issue.

File: DailyScreenshot/ModEntry.cs
Comment (CompSciLauren) [inline DailyScreenshot/ModEntry.cs:421]: Nice job drawing attention to this. Hopefully we'll be able to add the remaining config options before releasing the next update, in which case we should be able to remove this disclaimer. Otherwise we'll definitely want to include it.

File: DailyScreenshot/DailyScreenshot.csproj
Comment (CompSciLauren) [inline DailyScreenshot/DailyScreenshot.csproj:50]: Nit: Typo, dependencies. Not worth requesting a change (unnecessary delay), can fix while working on other changes before releasing next update.

File: DailyScreenshot/DailyScreenshot.csproj
Comment (CompSciLauren) [inline DailyScreenshot/DailyScreenshot.csproj:61]: Think this may have been added automatically or by accident, isn't needed here. Going to do same thing as with the typo; will fix later since it's minor and not worth delaying the PR getting merged.

**Review by CompSciLauren — APPROVED** [review]
Looks awesome! Tested manually, everything seems to behave as intended. Nice work 👍

**Comment by CompSciLauren** [comment]
Finished reviewing, looks great and is now merged! Thanks again for your efforts. Decided that I'll go ahead and see if I can add the other config options before we release the next update. Added issue #48 to help track this.

**Comment by m4ny** [comment]
> 3. Sorry I don't think I understand what you're referring to here. Can you please explain a bit more? What constant default values are you referring to, and what do you mean by "settable within range"?

To be honest, I don't know what I meant.  I think we can ignore this point.

**Pull request merged**
